database-migration/src/main/java/uk/ac/ebi/eva/dbmigration/mongodb/ExtractAnnotationFromVariant.java

```diff
@@ -0,0 +1,312 @@
+        Set<Integer> soSet = new TreeSet<>();
+
+        List<Document> cts = (List<Document>) originalAnnotField.get(outerField);
+        if (cts != null) {
+            for (Document ct : cts) {
+                Object sos = ct.get(innerField);
+                if (sos != null) {
+                    soSet.addAll((List<Integer>) sos);
+                }
+            }
+        }
+
+        return soSet;
+    }
+
+    private static Set<String> computeXrefSet(Document originalAnnotField, String outerField, String innerField) {
+        Set<String> xrefSet = new TreeSet<>();
+
+        List<Document> cts = (List<Document>) originalAnnotField.get(outerField);
+        if (cts != null) {
+            for (Document ct : cts) {
+                String xref = ct.getString(innerField);
+                if (xref != null) {
+                    xrefSet.add(xref);
+                }
+            }
+        }
+
+        return xrefSet;
+    }
+
+    private static List<Double> computeMinAndMaxScore(Document originalAnnotField, String field) {
+        Double min = Double.POSITIVE_INFINITY;
+        Double max = Double.NEGATIVE_INFINITY;
+        boolean thereIsAtLeastOneScore = false;
+
+        List<Document> cts = (List<Document>) originalAnnotField.get(CONSEQUENCE_TYPE_FIELD);
+        if (cts != null) {
+            for (Document ct : cts) {
+                Document document = ((Document) ct.get(field));
+                if (document != null) {
+                    Double score = (Double) document.get(SCORE_FIELD);
+                    if (score != null) {
+                        min = Math.min(min, score);
+                        max = Math.max(max, score);
+                        thereIsAtLeastOneScore = true;
+                    }
+                }
+            }
+        }
+        if (thereIsAtLeastOneScore) {
+            return Arrays.asList(min, max);
+        } else {
+            return Collections.emptyList();
+        }
+    }
+
+    @ChangeSet(order = "003", id = "updateAnnotationMetadata", author = "EVA")
+    public void updateAnnotationMetadata(MongoDatabase mongoDatabase) {
+        final MongoCollection<Document> annotationMetadataCollection = mongoDatabase.getCollection(
+                databaseParameters.getDbCollectionsAnnotationMetadataName());
+        logger.info("3) update annotation metadata in collection {}", annotationMetadataCollection.getNamespace());
+
+        String id = databaseParameters.getVepVersion() + "_" + databaseParameters.getVepCacheVersion();
+        Document metadata = new Document(ID_FIELD, id);
+        if (annotationMetadataCollection.count(metadata) == 0) {
+            metadata.append(VEP_VERSION_FIELD, databaseParameters.getVepVersion())
+                    .append(CACHE_VERSION_FIELD, databaseParameters.getVepCacheVersion());
+
+            annotationMetadataCollection.insertOne(metadata);
+        }
+    }
+}
```

Comment:
even though in the variants collection the old and new indexes are the same (annot.so and annot.xrefs, but now annot is an array), we are dropping those 2 anyways to improve the update performance, as we query by variants._id